content/developer/howtos/upgrade_custom_db.rst

```diff
@@ -0,0 +1,199 @@
+:show-content:
+:show-toc:
+
+.. _upgrade/upgrade_custom_db:
+
+===============================
+Upgrading a customized database
+===============================
+
+
+Upgrading to a new version of Odoo can be challenging, 
+especially if the database you work on contains custom modules.  
+This page intent is to explain the technical process of upgrading a database with customized modules.
+For numerous years, Odoo has followed this process, reaping the rewards of continual improvements along the way.
+
+For a standard overview, please refer to the
+:doc:`upgrade documentation </administration/upgrade>`.
+
+We consider a custom module, any module that extends the standard code of Odoo and that was not built with studio. 
+Before upgrading those modules, or before asking for them to be upgraded, 
+you might want to have a look at the :ref:`Service-level Agreement <upgrade/sla>` to make sure who's responsible for it.
+
+While you work on what we refer to as the **custom upgrade** of your database, 
+you must keep in mind what the goals of an upgrade are:
+
+#. Stay supported
+#. Get the latest features
+#. Enjoy the performance improvement
+#. Reduce the technical debt
+#. Benefit from security improvements
+
+
+With every new version of Odoo comes a bunch of changes,
+it can be addition but sometimes it also contains refactorization.
+All those changes can impact modules on which you have previously developed customization 
+which is why you will need to adapt them.
+This is why upgrading a database that contains custom modules requires additional steps compared to 
+a standard upgrade of an Odoo database since the source code of the custom modules
+must be upgraded as well. 
+
+Here are the steps we follow at Odoo to upgrade such databases:
+
+#. :ref:`Stop the devolopments and challenge them <upgrade/stop_developments>`  .
+#. :ref:`Request an upgraded database <upgrade/request_upgrade>`.
+#. :ref:`Make your module installable on an empty database <upgrade/empty_database>`.
+#. :ref:`Make your module installable on the upgraded database <upgrade/upgraded_database>`.
+#. :ref:`Test extensively and do a rehearsal <upgrade/testing_rehearsal>`.
+#. :ref:`Upgrade in production <upgrade/production>`.
+
+
+.. _upgrade/stop_developments:
+
+Stop the developments
+=============================
+
+Upgrade can be a tough process, starting it requires commitment and will block a significant amount of your development resources.
+If you decide to keep going for developments during that process,
+you will inevitably have to block resources for a longer time to work on the upgrade.
+Those features will need to be re-upgraded and tested, everytime you change them.
+
+This is why we recommend a complete freeze of the codebase when starting the upgrade process.
+Of course, you can still work on maintenance but developing new features is not productive.
```

Comment:
Maybe `bugfix` instead of maintenance?
That how we call it but IMO, "bugfix" is more understandable and means a smaller scope.
```suggestion
Of course, you can still work on bigfix but developing new features is not productive.
```